src/utils/authUtils.ts

```diff
@@ -6,12 +6,22 @@ import {
 import fbApp from '../database/clientApp';
 import { AuthDispatch } from '../screens/auth/AuthContext';
 
+/**
+ * Helper function to set the error message in the auth context.
+ * Use this function instead of using the dispatch directly.
+ * We can add more error handling/reporting logic (e.g. reporting errors to Sentry) here in the future.
+ */
+export const setAuthErrorMessage = (
+  dispatch: AuthDispatch,
+  errorMessage: string,
+) => dispatch({ type: 'SET_ERROR_MESSAGE', errorMessage });
+
 export const signIn = async (
   dispatch: AuthDispatch,
   params: { email: string; password: string },
 ) => {
   const auth = getAuth(fbApp);
-  signInWithEmailAndPassword(auth, params.email, params.password)
+  await signInWithEmailAndPassword(auth, params.email, params.password)
```

Comment:
@oahnh adding these `await`s fix the lag you discovered.

in case you're curious, since this wasn't being waited, the `SIGN_IN` dispatch was being triggered _after_ the `setShowErrorMessage(true)` in LoginScreen, so we were showing the old error message before it was updated to null (through the login dispatch).